File: quizz-service/src/main/java/com/app/quizzservice/request/response/ReviewResponse.java
package com.app.quizzservice.request.response;

import com.app.quizzservice.request.dto.ReviewDTO;

import java.util.List;

public record ReviewResponse(
        long testId,
        boolean hasMonitor,
        String name,
        long testAttemptId,
        int numberOfWarning,
        int totalCorrect,
        int score,
        String startTime,
        String endTime,
        int totalQuestions,
        List<ReviewQuestionResponse> questions
) {
    public ReviewResponse(List<ReviewDTO> dtos) {
        this(
                dtos.getFirst().testId(),
                dtos.getFirst().hasMonitor(),
                dtos.getFirst().name(),
                dtos.getFirst().testAttemptId(),
                dtos.getFirst().numberOfWarning(),
                dtos.getFirst().totalCorrect(),
                dtos.getFirst().score(),
                dtos.getFirst().startTime(),
                dtos.getFirst().endTime(),
                dtos.getFirst().totalQuestions(),
                dtos.stream().map(ReviewQuestionResponse::new).toList()
        );
    }

    public ReviewResponse() {
        this(0, false, "", 0, 0, 0, 0, "", "", 0, List.of());
    }
}

File: quizz-ui/src/app/shared/model/ReviewResponse.ts
export class ReviewResponse {
  constructor(
    public testId: number = 0,
    public hasMonitor: boolean = false,
    public name: string = '',
    public testAttemptId: number = 0,
    public numberOfWarning: number = 0,
    public totalCorrect: number = 0,
    public score: number = 0,
    public time: number = 0,
    public startTime: string = '',
    public endTime: string = '',
    public totalQuestions: number = 0,
    public percentCorrect: number = 0,
    public questions: ReviewQuestion[] = []
  ) {
  }
}

export class ReviewQuestion {
  constructor(
    public questionId: number = 0,
    public userAnswer: string = '',
    public isCorrect: boolean = false,
    public no: number = 0,
    public image: string = '',
    public content: string = '',
    public optionA: string = '',
    public optionB: string = '',
    public optionC: string = '',
    public optionD: string = '',
    public correctAnswer: string = ''
  ) {
  }
}
